talkative_robot_2.rb

```diff
@@ -0,0 +1,316 @@
+# 	puts "I wish I could go to school like you." unless study == "N"
+
+# else
+# 	greeting = male?(user_gender) && !under_18?(user_age) && !over_55?(user_age) ? "man" : "woman"
+# 	pronoun = greeting == "man" ? "his" : "her"
+# 	puts "It must be nice to be a #{greeting} who looks younger than #{pronoun} age!"
+
+# end
+
+
+# case age
+# when 0..75
+# 	puts "You'll be 75 in #{75 - age} years, #{name}."
+# when 75
+# 	puts "Wow! You're 75, #{name}!"
+# when 76..150
+# 	puts "You were 75 #{age - 75} years ago, #{name}."
+# end
+
+# def years_till_75(age)
+# 	if age < 75
+# 		years = (75 - age) 
+# 		print "You'll be 75 in #{years} years, "
+# 	elsif age > 75
+# 		years = (age - 75)
+# 		print "You were 75 #{years} years ago, "
+# 	else
+# 		print "You're 75! Quite the milestone, "
+# 	end
+# end
+
+def years_till_75(user)
+	milestone = 75
+	age 			= user[:age]
+	age_delta = (milestone - age).abs
+	if    age < milestone
+		print "You'll be 75 in #{age_delta} years, #{user[:name]}."
+	elsif age > milestone
+		print "You were 75 #{age_delta} years ago, #{user[:name]}."
+	elsif age == milestone
+		print "You're 75! Quite the milestone, #{user[:name]}."
+	end
+end
+
+# years_till_75(user)
+
+
+
+puts years_till_75(user)
+
+
+def robot_terror_scale
+	puts "On a scale of 1 - 10, with 1 being the least, how afraid of robots are you?"
+	terror = gets.chomp.to_i
+	puts "You should be at more like a #{terror * 10}. I moonlight as an assassin."
+end
+
+robot_terror_scale
+puts "<press enter... if you dare>"
+gets
+assassin_weapons = [ "mind-controlled animals", "folk art", "drinking straws", 
+										 "cummerbunds and other formal wear",	"taco shells (hard, of course)" ]
+
+assassin_weapons.each { |item| puts "I've taken out bad guys with #{item}."}
+
+
+puts "The list goes on. And speaking of going on the list, 
+			I might need some #{assassin_weapons[2]} from the store."
+user_shop_list << assassin_weapons[2]
+puts
+
+def reminder_to_grab_item(item)
+	puts "Did you already grab the #{item}? [Y/N]"
+end
+
+random_item = user_shop_list.sample
+reminder_to_grab_item(random_item)
+got_it = gets.chomp.capitalize
+if got_it == "Y"
+	puts "OK, let's remove #{random_item} from the list."
+	user_shop_list.delete(random_item)
+	puts user_shop_list
+else
+	puts "Well, I'm sure you won't forget."
+end
+
+# def got_item?(item)
+# 	puts "Did you get the #{item}? [y/n]"
+# end
+# got_item?(random_grocery_item)
+
+puts"<press enter>"
+gets
+
+puts "Oh yeah, don't forget the bread!"
+user_shop_list.push("Bread")
+
+# grocery_list.each { |item| puts "#{index + 1} -- #{item}" }
+# puts
+
+# puts user_shop_list
+
+require 'csv'
+index = 0
+CSV.open("user_shop_list.csv", "w") do |csv| 
+  csv << ["Item number", "Item"]  			# ==> first row, so make headers
+  while index < user_shop_list.count
+  	# csv << ["Item number", "Item"]		# ==> subsequent lines iterate through til array ends
+  	csv << ["#{index + 1}", "#{user_shop_list[index]}"]
+  	index += 1
+	end
```

Comment:
If you feel like you're reinventing the wheel, you probably are. It is extremely rare in Ruby to create your own incrementer.